File: app/src/main/java/org/odddev/fantlab/home/HomeProvider.java
package org.odddev.fantlab.home;

import org.odddev.fantlab.core.di.Injector;
import org.odddev.fantlab.core.rx.ISchedulersResolver;
import org.odddev.fantlab.core.storage.StorageManager;

import javax.inject.Inject;

import rx.Single;

/**
 * @author kenrube
 * @since 11.12.16
 */

public class HomeProvider implements IHomeProvider {

    @Inject
    StorageManager storageManager;

    @Inject
    ISchedulersResolver schedulersResolver;

    public HomeProvider() {
        Injector.getAppComponent().inject(this);
    }

    @Override
    public Single<String> getUserName() {
        return Single.just(storageManager.loadUsername())
                .compose(schedulersResolver.applyDefaultSchedulers());
    }

    @Override
    public void clearCookie() {
        storageManager.clearCookie();
    }
}

File: app/src/main/kotlin/org/odddev/fantlab/home/HomeProvider.kt
package org.odddev.fantlab.home

import io.reactivex.Observable
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers
import org.odddev.fantlab.core.di.Injector
import org.odddev.fantlab.core.storage.StorageManager
import javax.inject.Inject

class HomeProvider : IHomeProvider {

	@Inject
	lateinit var storageManager: StorageManager

	init {
		Injector.getAppComponent().inject(this)
	}

	override fun getUserName() = Observable.just<String>(storageManager.loadUsername())
				.subscribeOn(Schedulers.io())
				.observeOn(AndroidSchedulers.mainThread())

	override fun clearCookie() {
		storageManager.clearCookie()
	}
}
